actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -496,22 +506,22 @@ def multiple_sources_tag_builder(type, sources)
 
           if sources.size > 1
             content_tag(type, options) do
-              safe_join sources.map { |source| tag("source", src: send("path_to_#{type}", source, skip_pipeline: skip_pipeline)) }
+              safe_join sources.map { |source| tag("source", src: resolve_asset_source(type, source, skip_pipeline)) }
             end
           else
-            options[:src] = send("path_to_#{type}", sources.first, skip_pipeline: skip_pipeline)
+            options[:src] = resolve_asset_source(type, sources.first, skip_pipeline)
             content_tag(type, nil, options)
           end
         end
 
-        def resolve_image_source(source, skip_pipeline)
+        def resolve_asset_source(asset_type, source, skip_pipeline)
           if source.is_a?(Symbol) || source.is_a?(String)
-            path_to_image(source, skip_pipeline: skip_pipeline)
+            send("path_to_#{asset_type}", source, skip_pipeline: skip_pipeline)
```

Comment:
Does calling [path_to_asset](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/AssetUrlHelper.html#method-i-path_to_asset) or [asset_path](https://edgeapi.rubyonrails.org/classes/ActionView/Helpers/AssetUrlHelper.html#method-i-asset_path) here net-out to the same behavior?

Reply:
I'm not 100% sure. `image_path` is defined as

```ruby
def image_path(source, options = {})
  path_to_asset(source, { type: :image }.merge!(options))
end
```

so maybe we could do

```suggestion
            path_to_asset(source, type: asset_type, skip_pipeline: skip_pipeline)
```